clippy_lints/src/strings.rs

```diff
@@ -438,27 +440,94 @@ declare_clippy_lint! {
 
 declare_lint_pass!(StringToString => [STRING_TO_STRING]);
 
+fn is_parent_map_like(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<rustc_span::Span> {
+    if let Some(parent_expr) = get_parent_expr(cx, expr)
+        && let ExprKind::MethodCall(name, _, _, parent_span) = parent_expr.kind
+        && name.ident.name == sym::map
+        && let Some(caller_def_id) = cx.typeck_results().type_dependent_def_id(parent_expr.hir_id)
+        && (clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Result)
+            || clippy_utils::is_diag_item_method(cx, caller_def_id, sym::Option)
+            || clippy_utils::is_diag_trait_item(cx, caller_def_id, sym::Iterator))
```

Comment:
I've seen something similar in another lint, I think. I can't remember which though. Maybe one of the `methods/` lints? Could you quickly grep for those checks and see if this could be extracted into a common clippy_utils function?

Reply:
I found one other occurrence but I guess it's enough for a common function so let's go. :)

Reply:
Ah nevermind, the other one needs an MSRV condition on `Option` and `Result` (in `methods/manual_inspect.rs`).

Reply:
Ok, that's too different. Thanks for checking!